tools/metrics/prometheus_reporter.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"strings"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// PrometheusReporter is translating events into Prometheus metrics
+type PrometheusReporter struct {
+	registry   prometheus.Registerer
+	metricVecs map[string]interface{}
+}
+
+// NewPrometheusReporter creates a new PrometheusReporter instance
+func NewPrometheusReporter(registry prometheus.Registerer) *PrometheusReporter {
+	return &PrometheusReporter{
+		registry:   registry,
+		metricVecs: map[string]interface{}{},
+	}
+}
+
+// WriteMetrics implements MetricWriter. Metrics are buffered
+// in batches not written immediately to rate limit api calls
```

Comment:
JOOI whose API are we talking about protecting here?

Reply:
I don't even know where I was copy pasting this from, I'll remove the unrelated bits.

Reply:
updated